### PR TITLE
chore: Migrate all instances of old vuelidate to new v2 syntax [CW-3274]

### PR DESCRIPTION
**Pull request by fayazara**
## Description

Removes all the old vuelidate syntax and replaced it with the new `useValidate` composable and the `v$` helper.

| Component                          | Path                                                         | Migrated                                      | Tested |
|------------------------------------|--------------------------------------------------------------|-----------------------------------------------|--------|
| Login page                         | app/javascript/v3/views/login/Index.vue                      | ✅                                             | ✅      |
| Custom Attributes settings page    | app/javascript/dashboard/components/CustomAttribute.vue      | ✅                                             | ✅      |
| Account settings page              | app/javascript/dashboard/routes/dashboard/settings/account/Index.vue | ✅                                             | ✅      |
| Add Account Modal                  | app/javascript/dashboard/components/layout/sidebarComponents/AddAccountModal.vue | ✅                                             | ✅      |
| AICTA Modal                        | app/javascript/dashboard/components/widgets/AICTAModal.vue   | ✅                                             | ✅      |
| Conversation Advanced Filters      | app/javascript/dashboard/components/widgets/conversation/ConversationAdvancedFilter.vue | deprecated `$each` prop in validations object |        |
| Email Transript Modal              | app/javascript/dashboard/components/widgets/conversation/EmailTranscriptModal.vue | ✅                                             | ✅      |
| Linear Create Issue                | app/javascript/dashboard/components/widgets/conversation/linear/CreateIssue.vue | ✅                                             | ✅      |
| Template Parser                    | app/javascript/dashboard/components/widgets/conversation/WhatsappTemplates/TemplateParser.vue | ✅                                             |        |
| Delete Confirmation Modal          | app/javascript/dashboard/components/widgets/modal/ConfirmDeleteModal.vue | ✅                                             | ✅      |
| Add Custom Attribute               | app/javascript/dashboard/modules/contact/components/AddCustomAttribute.vue | ✅                                             | ✅      |
| Merge Contacts                     | app/javascript/dashboard/modules/contact/components/MergeContact.vue | ✅                                             | ✅      |
| Contacts Advanced Filters          | app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue | deprecated `$each` prop in validations object |        |
| Contact Form                       | app/javascript/dashboard/routes/dashboard/conversation/contact/ContactForm.vue | ✅                                             | ✅      |
| Conversation Form                  | app/javascript/dashboard/routes/dashboard/conversation/contact/ConversationForm.vue | ✅                                             | ✅      |
| Add Custom Views                   | app/javascript/dashboard/routes/dashboard/customviews/AddCustomViews.vue | ✅                                             | ✅      |
| Add Locale                         | app/javascript/dashboard/routes/dashboard/helpcenter/components/AddLocale.vue | ✅                                             | ✅      |
| Portal Settings Basic Form         | /app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsBasicForm.vue | ✅                                             | ✅      |
| Portal Settings Customization Form | /app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalSettingsCustomizationForm.vue | ✅                                             | ✅      |
| Add Category                       | app/javascript/dashboard/routes/dashboard/helpcenter/pages/categories/AddCategory.vue | ✅                                             | ✅      |
| Edit Category                      | app/javascript/dashboard/routes/dashboard/helpcenter/pages/categories/EditCategory.vue | ✅                                             | ✅      |
| CSML Bot Editor                    | app/javascript/dashboard/routes/dashboard/settings/agentBots/components/CSMLBotEditor.vue | ✅                                             | ✅      |
| Add Agent                          | app/javascript/dashboard/routes/dashboard/settings/agents/AddAgent.vue | ✅                                             | ✅      |
| Edit Agent                         | app/javascript/dashboard/routes/dashboard/settings/agents/EditAgent.vue | ✅                                             | ✅      |
| Add Attribute                      | app/javascript/dashboard/routes/dashboard/settings/attributes/AddAttribute.vue | ✅                                             | ✅      |
| Edit Attribute                     | app/javascript/dashboard/routes/dashboard/settings/attributes/EditAttribute.vue | ✅                                             | ✅      |
| Add Campaign                       | app/javascript/dashboard/routes/dashboard/settings/campaigns/AddCampaign.vue | ✅                                             | ✅      |
| Edit Campaign                      | app/javascript/dashboard/routes/dashboard/settings/campaigns/EditCampaign.vue | ✅                                             | ✅      |
| Add Canned                         | app/javascript/dashboard/routes/dashboard/settings/canned/AddCanned.vue | ✅                                             | ✅      |
| Edit Canned                        | app/javascript/dashboard/routes/dashboard/settings/canned/EditCanned.vue | ✅                                             | ✅      |
| IMAP Settings                      | app/javascript/dashboard/routes/dashboard/settings/inbox/ImapSettings.vue | ✅                                             | ✅      |
| SMTP Settings                      | app/javascript/dashboard/routes/dashboard/settings/inbox/SmtpSettings.vue | ✅                                             | ✅      |
| Widget Builder                     | app/javascript/dashboard/routes/dashboard/settings/inbox/WidgetBuilder.vue | ✅                                             | ✅      |
| 360 Dialog Whatsapp                | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/360DialogWhatsapp.vue | ✅                                             | ✅      |
| Inbox API settings                 | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Api.vue | ✅                                             | ✅      |
| SMS Bandwidth settings             | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/BandwidthSms.vue | ✅                                             | ✅      |
| Cloud Whatsapp Settings            | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/CloudWhatsapp.vue | ✅                                             | ✅      |
| Facebook Settings                  | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Facebook.vue | ✅                                             | ✅      |
| Line Settings                      | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Line.vue | ✅                                             | ✅      |
| Telegram Settings                  | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Telegram.vue | ✅                                             | ✅      |
| Twillio Settings                   | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Twilio.vue | ✅                                             | ✅      |
| Forward To option settings         | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/emailChannels/ForwardToOption.vue | ✅                                             | ✅      |
| Microsoft settings                 | app/javascript/dashboard/routes/dashboard/settings/inbox/channels/emailChannels/Microsoft.vue | ✅                                             | ✅      |
| Collaborators page                 | app/javascript/dashboard/routes/dashboard/settings/inbox/settingsPage/CollaboratorsPage.vue | ✅                                             | ✅      |
| Configuration Page                 | app/javascript/dashboard/routes/dashboard/settings/inbox/settingsPage/ConfigurationPage.vue | ✅                                             | ✅      |
| Dashboard App Modal Settings       | app/javascript/dashboard/routes/dashboard/settings/integrations/DashboardApps/DashboardAppModal.vue | ✅                                             | ✅      |
| Settings - Webhook Form            | app/javascript/dashboard/routes/dashboard/settings/integrations/Webhooks/WebhookForm.vue | ✅                                             | ✅      |
| Macro Form                         | app/javascript/dashboard/routes/dashboard/settings/macros/MacroForm.vue | deprecated `$each` prop in validations object |        |
| Change Password                    | app/javascript/dashboard/routes/dashboard/settings/profile/ChangePassword.vue | ✅                                             | ✅      |
| settings - User Basic Details      | app/javascript/dashboard/routes/dashboard/settings/profile/UserBasicDetails.vue | ✅                                             | ✅      |
| Password Edit                      | app/javascript/v3/views/auth/password/Edit.vue               | ✅                                             | ✅      |
| Password Reset form                | app/javascript/v3/views/auth/reset/password/Index.vue        | ✅                                             | ✅      |
| Signup form                        | app/javascript/v3/views/auth/signup/components/Signup/Form.vue | ✅                                             | ✅      |
| Login form                         | app/javascript/v3/views/login/Index.vue                      | ✅                                             | ✅      |
| Custom Attributes                         | app/javascript/dashboard/components/CustomAttribute.vue                      | ✅                                             | ✅      |
| Reply Email Head                         | app/javascript/dashboard/components/widgets/conversation/ReplyEmailHead.vue                      | ✅                                             | ✅      |
| Methods Mixin                        | app/javascript/dashboard/mixins/automations/methodsMixin.js                      | ✅                                             | ✅      |
| Validations mixin                        | app/javascript/dashboard/routes/dashboard/settings/labels/validationMixin.js                      | ✅                                             | ✅      |
| SLA Form                        | app/javascript/dashboard/routes/dashboard/settings/sla/SlaForm.vue                      | ✅                                             | ✅      |
| SLA Time Input                        | app/javascript/dashboard/routes/dashboard/settings/sla/SlaTimeInput.vue                      | ✅                                             | ✅      |
| SLA Validation Mixin                        | app/javascript/dashboard/routes/dashboard/settings/sla/validationMixin.js                      | ✅                                             | ✅      |
| Team Form                        | app/javascript/dashboard/routes/dashboard/settings/teams/TeamForm.vue                      | ✅                                             | ✅      |
| Add Agents                        | app/javascript/dashboard/routes/dashboard/settings/teams/Create/AddAgents.vue                      | ✅                                             | ✅      |
| Edit Agents                        | app/javascript/dashboard/routes/dashboard/settings/teams/Edit/EditAgents.vue                     | ✅                                             | ✅      |

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Tested manually when updating each component with the new syntax

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by scmmishra — COMMENTED** [review]
Tested this on staging. Will review the code and get this merged soon

File: app/javascript/dashboard/routes/dashboard/settings/agents/AddAgent.vue
Comment (scmmishra) [inline app/javascript/dashboard/routes/dashboard/settings/agents/AddAgent.vue:1]: Seems like the template usage is not updated here

**Review by scmmishra — CHANGES_REQUESTED** [review]
@fayazara found these files where the usage is not updated

- [ ] app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/inbox/AddAgents.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/labels/AddLabel.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/labels/EditLabel.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/agents/AddAgent.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/macros/MacroForm.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/macros/MacroNode.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/macros/MacroProperties.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/sla/SlaTimeInput.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/sla/SlaForm.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/automation/AddAutomationRule.vue
- [ ] app/javascript/dashboard/routes/dashboard/settings/automation/EditAutomationRule.vue

**Comment by scmmishra** [comment]
# Known Issues

- [ ] Validations not working for Add and Edit Automation Rule
- [ ] Validations not working for Macro Forms

**Comment by scmmishra** [comment]
@pranavrajs @fayazara I've finished any pending changes, also tested it on staging. LGTM

**Review by scmmishra — APPROVED** [review]

File: app/javascript/dashboard/helper/validations.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    'snooze_conversation',
+    'resolve_conversation',
+    'remove_assigned_team',
+  ];
+
+  if (
+    !noParamActions.includes(action.action_name) &&
+    (!action.action_params || action.action_params.length === 0)
+  ) {
+    return 'Action parameters are required';
```
Comment (muhsin-k) [inline app/javascript/dashboard/helper/validations.js:132]: @scmmishra Not a blocker. It seems like we are showing all the error messages in English only. Of course, we can handle this in a separate PR.
cc @pranavrajs

Comment (pranavrajs) [reply app/javascript/dashboard/helper/validations.js:132]: Good catch.

Comment (scmmishra) [reply app/javascript/dashboard/helper/validations.js:132]: Fixing it!

Comment (scmmishra) [reply app/javascript/dashboard/helper/validations.js]: @muhsin-k these commits solve it 
- https://github.com/chatwoot/chatwoot/pull/9623/commits/a55ec03eb12630526185aef1d93bb56c48fc77e0
- https://github.com/chatwoot/chatwoot/pull/9623/commits/d3d2e361b211cb9bcdd9cc5d11b2225d1d8cd472

**Comment by scmmishra** [comment]
@pranavrajs Need your review, you can skim through, @muhsin-k @fayazara and I have test it through

File: app/javascript/dashboard/components/widgets/conversation/helpers/emailHeadHelper.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import emailValidator from 'vuelidate/lib/validators/email';
+import { email as emailValidator } from '@vuelidate/validators';
 
 export const validEmailsByComma = value => {
   if (!value.length) return true;
   const emails = value.replace(/\s+/g, '').split(',');
-  return emails.every(email => emailValidator(email));
+  return emails.every(email => emailValidator.$validator(email));
```
Comment (pranavrajs) [inline app/javascript/dashboard/components/widgets/conversation/helpers/emailHeadHelper.js:6]: 👀

File: app/javascript/dashboard/helper/validations.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  }
+
+  if (!filter.filter_operator) {
+    return FILTER_OPERATOR_REQUIRED;
+  }
+
+  if (
+    filter.filter_operator !== 'is_present' &&
+    filter.filter_operator !== 'is_not_present' &&
+    !filter.values
```
Comment (pranavrajs) [inline app/javascript/dashboard/helper/validations.js:35]: Empty array condition.

File: app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue [hunk 4]
```diff
@@ -314,20 +300,23 @@ export default {
...
-        applied_filters: this.appliedFilters.map(filter => ({
-          key: filter.attribute_key,
-          operator: filter.filter_operator,
-          query_operator: filter.query_operator,
-        })),
-      });
+      if (Object.keys(this.validationErrors).length === 0) {
+        this.$store.dispatch(
+          'contacts/setContactFilters',
+          JSON.parse(JSON.stringify(this.appliedFilters))
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue:309]: cloneObject

File: app/javascript/dashboard/components/widgets/conversation/helpers/emailHeadHelper.js (hunk 2 above)
Comment (scmmishra) [reply app/javascript/dashboard/components/widgets/conversation/helpers/emailHeadHelper.js:6]: [Default export](https://github.com/vuelidate/vuelidate/blob/next/packages/validators/src/index.js) from the package exports email like [this](https://github.com/vuelidate/vuelidate/blob/next/packages/validators/src/withMessages/email.js) with a $validator object. 

To use it without that, we need to import it differently. See [docs](https://vuelidate-next.netlify.app/validators.html#using-builtin-validators)

File: app/javascript/dashboard/routes/dashboard/settings/macros/MacroForm.vue [hunk 5]
```diff
@@ -109,19 +99,29 @@ export default {
...
+      this.v$.$touch();
+      if (this.v$.$invalid) return;
+
       this.$emit('submit', this.macro);
     },
     resetNode(index) {
-      this.$v.macro.actions.$each[index].$reset();
+      // delete that index specifically
+      // so that the next item does not get marked invalid
+      delete this.errors[`action_${index}`];
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/macros/MacroForm.vue:119]: const { [`action_${index}`]: _, ...restOfTheErrors} = this.errors
this.errors = restOfTheErrors.

File: app/javascript/packs/widget.js [hunk 6]
```diff
@@ -18,7 +17,7 @@ import { domPurifyConfig } from '../shared/helpers/HTMLSanitizer';
 const PhoneInput = () => import('../widget/components/Form/PhoneInput');
 
 Vue.use(VueI18n);
-Vue.use(Vuelidate);
+
```
Comment (pranavrajs) [inline app/javascript/packs/widget.js:20]: Test the widget functionalities like pre-chat form, form message (Chat Form)

Comment (pranavrajs) [reply app/javascript/packs/widget.js:20]: Email input message (Disable pre-chat form, the email input would appear on the first message)

**Review by pranavrajs — APPROVED** [review]
Please test the conditions mentioned above before merging.

File: app/javascript/dashboard/helper/validations.js (hunk 3 above)
Comment (scmmishra) [reply app/javascript/dashboard/helper/validations.js]: Added a check anyway

File: app/javascript/dashboard/routes/dashboard/settings/macros/MacroForm.vue (hunk 5 above)
Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/settings/macros/MacroForm.vue]: Done!

File: app/javascript/packs/widget.js (hunk 6 above)
Comment (scmmishra) [reply app/javascript/packs/widget.js:20]: Tested this

File: app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue (hunk 4 above)
Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/contacts/components/ContactsAdvancedFilters.vue:309]: Will do in a separate PR

**Pull request merged**